extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsController.class.php

```diff
@@ -28,37 +28,86 @@ public static function onBeforePageDisplay() {
 		return true;
 	}
 
-	public static function render( $input, array $args ) {
-		global $wgCityId;
+	/**
+	 * Checks arguments for errors.
+	 * @param array $args
+	 * @oaram errorMessage Return parameter with the proper error message to show. Disregard if return is false
+	 * @return true if ok, false if error
+	 */
+	private static function checkArguments( array $args, $modelData, &$errorMessage ) {
+		// mostrecent must be bool
+		if ( array_key_exists( 'mostrecent', $args ) && $args['mostrecent'] !== 'true' && $args['mostrecent'] !== 'false' ) {
+			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'mostrecent' )->plain() .
+				wfMessage( 'embeddable-discussions-parameter-error-boolean' )->plain();
+
+			return false;
+		}
 
-		$showLatest = !empty( $args['mostrecent'] ) && filter_var( $args['mostrecent'], FILTER_VALIDATE_BOOLEAN );
-		$itemCount = empty( $args['size'] ) ? self::ITEMS_DEFAULT : intval( $args['size'] );
-		$columns = empty( $args['columns'] ) ? self::COLUMNS_DEFAULT : intval( $args['columns'] );
-		$category = empty( $args['category'] ) ? '' :  $args['category'];
+		// size must be integer in range
+		if ( array_key_exists( 'size', $args ) ) {
+			$size = $args['size'];
 
-		if ( $itemCount > self::ITEMS_MAX ) {
-			$itemCount = self::ITEMS_MAX;
-		}
+			$errorMessage = wfMessage( 'embeddable-discussions-parameter-error', 'size' )->plain() .
+				wfMessage( 'embeddable-discussions-parameter-error-range',
+					self::ITEMS_MIN , self::ITEMS_MAX )->plain();
 
-		if ( $itemCount < self::ITEMS_MIN ) {
-			$itemCount = self::ITEMS_MIN;
+			if ( !ctype_digit( $size ) ) {
+				return false;
+			} else if ( intval( $size ) > self::ITEMS_MAX || intval( $size ) < self::ITEMS_MIN ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This branch's code block is the same as the block for the branch on line 54. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1871)